tests/admin/test_routes.py

```diff
@@ -0,0 +1,794 @@
+        self.assert_authenticated_request_calls(url, self.controller.lanes)
+        self.assert_supported_methods(url, 'GET', 'POST')
+    
+    def test_lane(self):
+        url = "/admin/lane/<lane_identifier>"
+        self.assert_authenticated_request_calls(
+            url, self.controller.lane, '<lane_identifier>', http_method='DELETE'
+        )
+        self.assert_supported_methods(url, 'DELETE')
+
+    def test_show_lane(self):
+        url = "/admin/lane/<lane_identifier>/show"
+        self.assert_authenticated_request_calls(
+            url, self.controller.show_lane, '<lane_identifier>', http_method='POST'
+        )
+        self.assert_supported_methods(url, 'POST')
+    
+    def test_hide_lane(self):
+        url = "/admin/lane/<lane_identifier>/hide"
+        self.assert_authenticated_request_calls(
+            url, self.controller.hide_lane, '<lane_identifier>', http_method='POST'
+        )
+        self.assert_supported_methods(url, 'POST')
+    
+    def test_reset(self):
+        url = "/admin/lanes/reset"
+        self.assert_authenticated_request_calls(
+            url, self.controller.reset, http_method='POST'
+        )
+        self.assert_supported_methods(url, 'POST')
+    
+    def test_change_order(self):
+        url = "/admin/lanes/change_order"
+        self.assert_authenticated_request_calls(
+            url, self.controller.change_order, http_method='POST'
+        )
+        self.assert_supported_methods(url, 'POST')
+
+class TestTimestamps(AdminRouteTest):
+
+    CONTROLLER_NAME = "timestamps_controller"
+
+    def test_diagnostics(self):
+        url = "/admin/diagnostics"
+        self.assert_authenticated_request_calls(url, self.controller.diagnostics)
+
+class TestAdminView(AdminRouteTest):
+
+    CONTROLLER_NAME = "admin_view_controller"
+
+    def test_admin_view(self):
+        url = "/admin/web/"
+        self.assert_request_calls(
+            url, self.controller, None, None, path=None
+        )
+
+        url = "/admin/web/collection/a/collection/book/a/book"
+        self.assert_request_calls(
+            url, self.controller, "a/collection", "a/book", path=None
+        )
+
+        url = "/admin/web/collection/a/collection"
+        self.assert_request_calls(
+            url, self.controller, "a/collection", None, path=None
+        )
+
+        url = "/admin/web/book/a/book"
+        self.assert_request_calls(
+            url, self.controller, None, "a/book", path=None
+        )
+
+        url = "/admin/web/a/path"
+        self.assert_request_calls(
+            url, self.controller, None, None, path="a/path"
+        )
+
+class TestAdminStatic(AdminRouteTest):
+
+    CONTROLLER_NAME = "static_files"
+
+    def test_static_file(self):
+        url = "/admin/static/circulation-web.js"
+
+        local_path = os.path.abspath(os.path.dirname(__file__))
+        local_path = local_path.split('/tests')[0] + "/api/admin/node_modules/simplified-circulation-web/dist"
```

Comment:
Is there a better way I can mock this or is this fine?

Reply:
Maybe something like this example from `test_controller.py`:

```
directory = os.path.join(os.path.abspath(os.path.dirname(__file__)), "..", "resources", "images")
```

Using ".." on the current directory is the key improvement I think.

Reply:
It's close but I'm not actually opening up a file so the `".."` are still there. Since it's doing a string comparison, it doesn't seem to help.

Reply:
I see. Calling `os.path.abspath` on a path that contains ".." will give you a path without the "..". Does that help?

Reply:
Yes, it does, thanks.